### ISSUE TITLE
Recurring job missing invocation after updating to 1.8.3

### ISSUE THREAD

**Issue by cezar-pimentel** [HangfireIO/Hangfire#2263]
Hello guys, today I noticed a strange behavior on my hangfire server.

I have 4 jobs scheduled each on a different period of time.

After updating the library to version 1.8.3 one of the recurring jobs stopped triggering automatically on it's scheduled time.

All other three jobs are working normally, only one is showing this problem.

Today I monitored it, and it showed: Next running time in 1 minute, and after that minute it didn't triggered.

Tomorrow I'll rollback to the previous version and I'll let you know if it starts triggering again.

No changes on the code were made, the only difference is really the library version. Before my change I was using version 1.7

Does anyone have any idea why this is happening?

Thanks

**Comment by odinserj** [comment]
Hm, can you show your calls to `AddOrUpdate` methods of your recurring jobs, highlighting the problematic one?

**Comment by cezar-pimentel** [comment]
Sorry about the delayed response. Yes, here it goes:

![image](https://github.com/HangfireIO/Hangfire/assets/12258489/1e757497-1731-4e5f-a80a-534f9d4c1a30)

**Comment by odinserj** [comment]
Yes, `Ignorable` mode was added in 1.8.X only and it was unavailable in 1.7.X version. So as far as I understand, you have problems with the recurring job that has `Ignorable` mode, while 3 of your other background jobs work fine and have non-ignorable mode set (or using default one)?

Can you tell me what cron expression is used for that problematic recurring job, and what's your `SchedulePollingInterval` set in the `BackgroundJobServerOptions` (or you are using the default value here)?

**Comment by odinserj** [comment]
Can you also check the exact version of the `Hangfire.Core` assembly? There were important fixes in 1.8.3 and we shoud ensure your application aren't using 1.8.0–1.8.2 somehow.

**Comment by cezar-pimentel** [comment]
> Yes, `Ignorable` mode was added in 1.8.X only and it was unavailable in 1.7.X version. So as far as I understand, you have problems with the recurring job that has `Ignorable` mode, while 3 of your other background jobs work fine and have non-ignorable mode set (or using default one)?
> 
> Can you tell me what cron expression is used for that problematic recurring job, and what's your `SchedulePollingInterval` set in the `BackgroundJobServerOptions` (or you are using the default value here)?

This is the problematic cron expression: "30 1 * * *" which stands for Everyday at 01:30.

For the SchedulePollingInterval, I'm not setting it, so I'm using the default value.

And for the Hangfire.Core version this is the one I'm using right now:
![image](https://github.com/HangfireIO/Hangfire/assets/12258489/47a13d1e-a355-48e3-91d5-6dcc769390ed)

But let's do this. I'll deploy the application once again with the CI package version, and let's see how it will run today, because I don't recall which version of the Hangfire.Core I was using the day I noticed the missfire.

I'll follow up today at 01:30 to check if there was a missfire or not, and I'll let you know ASAP.

Thanks my friend!

**Comment by cezar-pimentel** [comment]
Yeah, I can confirm, it misfired with the latest version. 🥲

**Comment by odinserj** [comment]
Any chance that your application process wasn't working that time? If it was started after that time, `Ignorable` mode will disallow to run a missed recurring job that's by design. What happens if you set the `Relaxed` mode for that background job, is there any difference between the scheduled time and execution time of a background job (this will help us to diagnose whether process starts a bit later)?

**Comment by cezar-pimentel** [comment]
No, not a change. All 4 background jobs are hosted in the same WebApplication which is a self host executable installed as a Windows Service, so that it will be always running.

I'll change it to relaxed once I get home and let's see what happens today at 01:30.

**Comment by cezar-pimentel** [comment]
I've changed the MisfireHandling to Relaxed on all jobs.

![image](https://github.com/HangfireIO/Hangfire/assets/12258489/3b7f113e-5b97-467d-ba86-e89bcbfc18dd)

Now I'll wait until 01:30AM to see if it will trigger and I'll let you know ASAP.

Thanks

**Comment by cezar-pimentel** [comment]
Confirmed. The job triggered correctly after changing to Relaxed.

That saif, I believe there still issues regarding the usage of ignorable.

Let me know if you need any other testing config.

**Comment by odinserj** [comment]
Is it possible to go to the Job Details page of the corresponding execution and send me the "Time" job parameter's value?

![image](https://github.com/HangfireIO/Hangfire/assets/1078718/113785ef-f316-4573-8dbc-11c27418ac88)

I will deserialize it and compare with the scheduled time. If there is a big difference between the values (so scheduled time does not match the actual time), the probably the process was inactive. If not, and values are close to each other, then this is definitely a bug.

**Comment by cezar-pimentel** [comment]
For sure! Here it is!

![image](https://github.com/HangfireIO/Hangfire/assets/12258489/285796b4-a604-4231-87c1-ce37cbcad51d)

I can assure that the process was active.

Would mind telling me how are going to deserialize this time parameter?
